scripts/pylib/twister/twisterlib/testsuite.py

```diff
@@ -310,7 +310,13 @@ def scan_testsuite_path(testsuite_path):
         except ValueError as e:
             logger.error("%s: error parsing source file: %s" % (filename, e))
 
+    src_dir_pathlib_path = Path(src_dir_path)
     for filename in find_c_files_in(testsuite_path):
+        # If we have already scanned those files in the src_dir step, skip them.
+        filename_path = Path(filename)
+        if src_dir_pathlib_path in filename_path.parents:
+            continue
+
```

Comment:
@LukaszMrugala please check this one in my PR which I mentioned yesterdy https://github.com/zephyrproject-rtos/zephyr/pull/71541/commits/2b63fd087f9ab24dba5385eff9ad9dafabccd226 , you implement is not safe, the path may not be in the exact parents, they maybe offtree, so match the file path only is safer